### PR TITLE
Added support for non-personalized ads

### PR DESCRIPTION
**Pull request by Skyost**
**Motivation :**

Quoting from [Google](https://developers.google.com/admob/unity/eu-consent) :

> Under the Google EU User Consent Policy, you must make certain disclosures to your users in the European Economic Area (EEA) and obtain their consent to use cookies or other local storage, where legally required, and to use personal data (such as AdID) to serve ads. This policy reflects the requirements of the EU ePrivacy Directive and the General Data Protection Regulation (GDPR).

> Ads served by Google can be categorized as personalized or non-personalized, both requiring consent from users in the EEA. By default, ad requests to Google serve personalized ads, with ad selection based on the user's previously collected data. Google also supports configuring ad requests to serve non-personalized ads.

Therefore, in order to comply with General Data Protection Regulation law in EU, we need to ask whether users are okay to receive personalized ads or non-personalized ads (NPA). I think it would be a nice feature to integrate into this plugin as we could use it alongside [admob_consent](https://pub.dev/packages/admob_consent) for example.

**This pull request :**

* Adds the `nonPersonalizedAds` boolean argument in banner, interstitial and reward ads.
* Does a little bit of clean up in the repetitive channel methods arguments (see [this commit](https://github.com/kmcgill88/admob_flutter/pull/215/commits/4a49d9c93d3b843d045fb52d1f53caed481e7ec9)).

### CODE REVIEW

**Comment by kmcgill88** [comment]
@Skyost you are crushing it, man! Thanks for your contributions. I will review this ASAP.

To be clear. Are you suggesting to add `admob_consent` as a dependency of this package, `admob_flutter`?

**Comment by Skyost** [comment]
> you are crushing it, man! Thanks for your contributions. I will review this ASAP.

Hey, no problem ! I really love this plugin and I think it's part of the open-source philosophy to contribute to what you like.

> To be clear. Are you suggesting to add `admob_consent` as a dependency of this package, `admob_flutter` ?

No sorry, perhaps I expressed myself badly, but it was more of an example usage.

File: lib/src/admob_banner.dart [hunk 1]
```diff
@@ -11,13 +11,15 @@ class AdmobBanner extends StatefulWidget {
   final AdmobBannerSize adSize;
   final void Function(AdmobAdEvent, Map<String, dynamic>) listener;
   final void Function(AdmobBannerController) onBannerCreated;
+  final bool nonPersonalizedAds;
```
Comment (kmcgill88) [inline lib/src/admob_banner.dart:14]: Not a deal-breaker but would be nice to have more docs on this, and a link. https://support.google.com/admob/answer/7676680

Comment (kmcgill88) [reply lib/src/admob_banner.dart:14]: Also adding to the README the use of [admob_consent](https://pub.dev/packages/admob_consent) as a nice way to get user's consent.

File: ios/Classes/AdmobRewardPlugin.swift [hunk 2]
```diff
@@ -88,10 +88,17 @@ public class AdmobRewardPlugin: NSObject, FlutterPlugin {
...
     
-    private func loadRewardBasedVideoAd(id: Int, rewardBasedVideoAdUnitId: String) {
+    private func loadRewardBasedVideoAd(id: Int, rewardBasedVideoAdUnitId: String, nonPersonalizedAds: Bool) {
         let video = GADRewardedAd(adUnitID: rewardBasedVideoAdUnitId)
         rewardAds[id] = video
         let request = GADRequest()
+
+        if (nonPersonalizedAds) {
+            let extras = GADExtras()
+            extras.additionalParameters = ["npa": "1"]
```
Comment (kmcgill88) [inline ios/Classes/AdmobRewardPlugin.swift:98]: Can you please share a link to `npa` additional parameters docs.

I'm only seeing [Targeting](https://developers.google.com/admob/ios/targeting). It states 

> This parameter disables personalized advertising, including remarketing, for that specific ad request. It also disables requests to third-party ad vendors, such as ad measurement pixels and third-party ad servers.

Comment (Skyost) [reply ios/Classes/AdmobRewardPlugin.swift:98]: Sure, [here](https://developers.google.com/admob/android/eu-consent#forward-consent)'s where I found it.

File: lib/src/admob_banner.dart (hunk 1 above)
Comment (Skyost) [reply lib/src/admob_banner.dart:14]: I'm gonna do that !

File: ios/Classes/AdmobRewardPlugin.swift (hunk 2 above)
Comment (kmcgill88) [reply ios/Classes/AdmobRewardPlugin.swift:98]: This approach is marked as deprecated. Aaah, now I think I'm connecting dots. This [iOS](https://developers.google.com/admob/ump/ios/quick-start), [Android](https://developers.google.com/admob/ump/android/quick-start) is the new EU consent method. This appears to be a different framework than what `admob_consent` is using.

Comment (Skyost) [reply ios/Classes/AdmobRewardPlugin.swift:98]: Oh yes, as you said, the UMP SDK seems to be the new recommended way to handle user consent (although it seems to be [a bit messy](https://stackoverflow.com/questions/63464270/how-to-request-non-personalized-admob-ads-with-the-new-unified-messaging-platfor)). But the `npa` parameter is, and [will still be](https://groups.google.com/g/google-admob-ads-sdk/c/8hz_x8cIOzQ/m/IxKuBtcmAAAJ) supported for a long time.

**Comment by Skyost** [comment]
I've updated the README file according to your instructions and what you've found [here](https://github.com/kmcgill88/admob_flutter/pull/215#discussion_r482104896).

**Comment by kmcgill88** [comment]
I'll re-review this over the weekend. Thanks!

**Comment by joshmatta** [comment]
Any update on this merge

**Comment by Skyost** [comment]
@joshmatta If you want, I've created a little plugin using what @kmcgill88 found [here](https://github.com/kmcgill88/admob_flutter/pull/215#discussion_r482104896) : [flutter_funding_choices](https://pub.dev/packages/flutter_funding_choices). This pull request is meant to implement the legacy way in this plugin.

**Comment by deakjahn** [comment]
I found the README addition to be a bit confusing:

"Pass nonPersonalizedAds: false to the classes constructor (AdmobBanner, AdmobInterstitial and AdmobReward) in order to not display personalized ads for users who don't give their consent."

I should pass true, shouldn't I? False is the default value, so there is no point is passing it in, besides, those who didn't give their consent should receive non personalized ads all right. Or is the underlying code perhaps doing it the other way around?

**Comment by Skyost** [comment]
@deakjahn Oh yes, you're right ! I switched it in my mind when I wrote it, I don't what's happened 😄

**Pull request merged**